docs/sdk/src/reference_docs/frame_tokens.rs

```diff
@@ -0,0 +1,126 @@
+//! - The distinction between token traits and trait implementations in FRAME, and why this
+//!   distinction is helpful
+//! - Token-related traits avaliable in FRAME
+//! - Token-related trait implementations in FRAME
+//! - How to choose the right trait or trait implementation for your use case
+//! - Where to go next
+//!
+//! ## Getting Started
+//!
+//! The most ubiquitous way to add a token to a FRAME runtime with [`pallet_balances`]. Read
+//! more about pallets [here](crate::polkadot_sdk::frame_runtime#pallets).
+//!
+//! You may then write custom pallets that interact with [`pallet_balances`]. The fastest way to
+//! get started with that is by
+//! [tightly coupling](crate::reference_docs::frame_pallet_coupling#tight-coupling-pallets) your
+//! custom pallet to [`pallet_balances`].
+//!
+//! However, to keep pallets flexible and modular, it is often prefered to
+//! [loosely couple](crate::reference_docs::frame_pallet_coupling#loosely--coupling-pallets).
+//!
+//! To achieve
+//! [loose coupling](crate::reference_docs::frame_pallet_coupling#loosely--coupling-pallets),
+//! we seperate token logic into traits and trait implementations.
+//!
+//! ## Traits and Trait Implementations
+//!
+//! Broadly speaking, token logic in FRAME can be divided into two categories: traits and
+//! trait implementations.
+//!
+//! **Traits** define common interfaces that types of tokens should implement. For example, the
+//! [`frame_support::traits::fungible::Inspect`] trait specifies an interface for *inspecting*
+//! token state such as the total issuance of the token, the balance of individual accounts, etc.
+//!
+//! **Trait implementations** are concrete implementations of these traits. For example, one of the
+//! many traits [`pallet_balances`] implements is [`frame_support::traits::fungible::Inspect`]*.
+//!
+//! The distinction between traits and trait implementations is helpful because it allows pallets
+//! and other logic to be generic over their dependencies, avoiding tight coupling.
+//!
+//! To illustrate this with an example let's consider [`pallet_preimage`]. This pallet takes a
+//! deposit in exchange for storing some preimage for use later. A naive implementation of the
+//! pallet may use [`pallet_balances`] in a tightly coupled manner, directly calling methods
+//! on the pallet to reserve and unreserve deposits. This approach works well,
+//! until someone has a use case requiring that an asset from a different pallet such as
+//! [`pallet_assets`] is used for the deposit. Rather than tightly couple [`pallet_preimage`] to
+//! [`pallet_balances`], [`pallet_assets`], and every other token-handling pallet a user
+//! could possibly specify, [`pallet_preimage`] does not specify a concrete pallet as a dependency
+//! but instead accepts any dependency which implements the
+//! [`frame_support::traits::tokens::currency::ReservableCurrency`] trait. This allows
```

Comment:
```suggestion
//! [`frame_support::traits::tokens::currency::ReservableCurrency`] trait, namely via its [`pallet_preimage::Config::Currency`] associated type. This allows
```

Suggesting a concrete link to https://paritytech.github.io/polkadot-sdk/master/pallet_preimage/pallet/trait.Config.html#associatedtype.Currency